app/ghcup/BrickMain.hs

```diff
@@ -131,167 +319,159 @@ showKey Vty.KUp = "↑"
-
-        drawnElements = flip V.imap es $ \i' e ->
-          let addSeparator w = case es !? (i' - 1) of
-                Just e' | lTool e' /= lTool e ->
-                  hBorder <=> w
-                _                             -> w
-
-              isSelected  = Just i' == listSelected
-              elemWidget  = drawElem i' isSelected e
-              selItemAttr = if foc
-                then withDefAttr listSelectedFocusedAttr
-                else withDefAttr listSelectedAttr
-              makeVisible' = if isSelected then visible . selItemAttr else id
-          in  addSeparator $ makeVisible' elemWidget
-
-      in render
-        $ viewport "GHCup" Vertical
-        $ vBox
-        $ V.toList drawnElements
-
+            Just d  -> [Brick.withAttr (Brick.attrName "day") $ Brick.str (show d)])
 
 minHSize :: Int -> Widget n -> Widget n
-minHSize s' = hLimit s' . vLimit 1 . (<+> fill ' ')
+minHSize s' = Brick.hLimit s' . Brick.vLimit 1 . (<+> Brick.fill ' ')
 
-
-app :: AttrMap -> AttrMap -> App BrickState e String
+app :: AttrMap -> AttrMap -> App BrickState () Name
 app attrs dimAttrs =
-  App { appDraw         = \st -> [ui dimAttrs st]
-  , appHandleEvent  = \be -> get >>= \s -> eventHandler s be
-  , appStartEvent   = return ()
-  , appAttrMap      = const attrs
-  , appChooseCursor = showFirstCursor
-  }
+  App { appDraw         = drawUI dimAttrs
+      , appHandleEvent  = eventHandler
+      , appStartEvent   = return ()
+      , appAttrMap      = const attrs
+      , appChooseCursor = Brick.showFirstCursor
+      }
+
+
+drawUI :: AttrMap -> BrickState -> [Widget Name]
+drawUI dimAttrs st = 
+  case st ^. mode of
+    Navigation -> [ui dimAttrs st]
+    Tutorial   -> 
+      let tutorial = centerLayer
+                   $ Brick.hLimitPercent 75
+                   $ Brick.vLimitPercent 50
+                   $ Brick.withBorderStyle unicode
+                   $ borderWithLabel (Brick.txt "Tutorial")
+                   $ Brick.vBox 
+                     [ Brick.txt "GHCup is a tool for managing your Haskell tooling."
+                     , center (Brick.txt "--- o ---")
+                     , Brick.txt "This symbol "
+                        <+> Brick.withAttr (Brick.attrName "installed") (Brick.str "✓ ")
+                        <+> Brick.txt " means that the tool is installed but not in used"
+                     , Brick.txt "This symbol "
+                        <+> Brick.withAttr (Brick.attrName "set") (Brick.str "✔✔") 
+                        <+> Brick.txt " means that the tool is installed and in used"
+                     , Brick.txt "This symbol "
+                        <+> Brick.withAttr (Brick.attrName "not-installed") (Brick.str "✗ ")
+                        <+> Brick.txt " means that the tool isn't installed"
+                     , center (Brick.txt "--- o ---")
+                     , Brick.txt "Press Enter to exit the tutorial"
+                     ]
+       in [tutorial, ui dimAttrs st]
+
```

Comment:
`drawUI` now splits, depending on which mode we are. This is the reason `Mode` type is so usefull. We can define handlers, drawings, etc... for each mode with barely extra complexity